src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -77,6 +84,23 @@ public final boolean init(Literal<?>[] args, int matchedPattern, ParseResult par
 			throw new IllegalStateException();
 		skriptEventInfo = (SkriptEventInfo<?>) syntaxElementInfo;
 
+		// evaluate whether this event supports listening to cancelled events
+		supportsListeningBehavior = false;
+		for (Class<? extends Event> eventClass : getEventClasses()) {
+			if (Cancellable.class.isAssignableFrom(eventClass)) {
+				supportsListeningBehavior = true;
+				break;
+			}
+		}
+
+		listeningBehavior = eventData.getListenerBehavior();
+		// if the behavior is non-null, it was set by the user
+		if (listeningBehavior != null && !isListeningBehaviorSupported()) {
+			String eventName = skriptEventInfo.name.toLowerCase(Locale.ENGLISH);
+			Skript.error(Utils.A(eventName) + " event does not support listening for cancelled or uncancelled events.");
```

Comment:
I think it'd be better to just say that the event does not support a custom listening behavior. Though i wonder if "on any \<event\>" should really error if listening behavior isn't supported

Reply:
I was a little worried that users wouldn't really understand what that means, but it would be more succinct. Let me know if you think that my worries are unfounded.

For the `any`, I agree, I'll add that.

Reply:
Actually, I take back the `any` thing. I think that if the listening behavior is not supported, you shouldn't be able to listen to any. If an event overrides isSupported to deny listening behaviors, allowing `any` could get around that. Let me know your thoughts.